core/src/main/scala/ai/h2o/sparkling/H2OFrame.scala

```diff
@@ -106,6 +108,85 @@ class H2OFrame private (
       H2OFrame(rapidsFrameV3.key.name)
     }
   }
+
+  /**
+    * Left join this frame with another frame
+    *
+    * @param another right frame
+    * @return new frame
+    */
+  def leftJoin(another: H2OFrame): H2OFrame =
+    join(another, allFromCurrent = true, allFromAnother = false, "radix")
+
+  /**
+    * Right join this frame with another frame
+    *
+    * @param another right frame
+    * @return new frame
+    */
+  def rightJoin(another: H2OFrame, method: String = "AUTO"): H2OFrame = {
+    // Right join fails under "radix". The other variant is "hash" method
+    // but that method does not support strings in columns and does not work correctly if there are duplicate
+    // columns in the frame. Use Spark for now
+    joinUsingSpark(another, "right")
+  }
+
+  /**
+    * Inner join this frame with another frame
+    *
+    * @param another right frame
+    * @return new frame
+    */
+  def innerJoin(another: H2OFrame): H2OFrame =
+    join(another, allFromCurrent = false, allFromAnother = false, "radix")
+
+  /**
+    * Outer join this frame with another frame
+    *
+    * @param another right frame
+    * @return new frame
+    */
+  def outerJoin(another: H2OFrame): H2OFrame = {
+    // Outer join is broken in H2O, simulate H2O's join via Spark for now
```

Comment:
Is there any Jira on H2O-3 side for this?

Reply:
I want to first discuss with someone from H2O before creating one.

Reply:
ok

Reply:
Discussing with @wendycwong